### PR TITLE
[mds-types] [most other packages] Switch to string literal arrays for states & events

### PR DESCRIPTION
**Pull request by avatarneil**
## 📚 Purpose
Switching to string literal ararys as opposed to Enums makes type composition much easier, and is a prerequisite for adding additional state machines to mds-core. This PR switches the state and event types from using our custom `Enum` object to const assertions, see the [Typescript 3.4 Release Notes](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-4.html#const-assertions)

## 📦 Impacts:
mds-types
most other packages

## ✅ PR Checklist
- [x] `yarn build` passing for whole repo
- [x] `yarn test` passing in mds-agency

### CODE REVIEW

File: packages/mds-agency-cache/index.ts [hunk 1]
```diff
@@ -408,7 +408,7 @@ async function readDevicesStatus(query: {
       try {
         const parsedItem = parseEvent(item)
         if (
-          parsedItem.vehicle_state === VEHICLE_STATES.removed ||
+          parsedItem.vehicle_state === 'removed' ||
```
Comment (karcass) [inline packages/mds-agency-cache/index.ts:411]: Is there a compelling reason not to have constants like VS_REMOVED?

File: packages/mds-utils/state-machine.ts [hunk 2]
```diff
@@ -147,7 +131,7 @@ const generateTransitionLabel = (status: VEHICLE_STATE, nextStatus: VEHICLE_STAT
 // Punch this output into http://www.webgraphviz.com/
 const generateGraph = () => {
   const graphEntries = []
-  const statuses: VEHICLE_STATE[] = Object.values(VEHICLE_STATES)
+  const statuses: Readonly<VEHICLE_STATE[]> = VEHICLE_STATES
```
Comment (karcass) [inline packages/mds-utils/state-machine.ts:134]: TS voodoo!

File: packages/mds-agency-cache/index.ts (hunk 1 above)
Comment (avatarneil) [reply packages/mds-agency-cache/index.ts:411]: Just one more thing to maintain ¯\_(ツ)_/¯

File: packages/mds-types/index.ts [hunk 3]
```diff
@@ -93,9 +93,9 @@ export const VEHICLE_EVENTS = Enum(
   'trip_leave_jurisdiction',
   'trip_start',
   'unspecified'
-)
+] as const
```
Comment (janedotx) [inline packages/mds-types/index.ts:96]: 👍 for never having to type `Object.keys(VEHICLE_EVENTS)` again.

**Pull request merged**